### PR TITLE
Null check for BingMapKey

### PR DESCRIPTION
**Pull request by swetha571**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

# Description of the change
BingMpakey null value check
...

# Motivation for the change

...

### CODE REVIEW

File: src/components/pages/dashboard/dashboard.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ class DashboardPage extends Component {
           <Grid fluid className="layout">
             <Row>
               <Col md={8}>
-                <DeviceMap {...deviceMapProps} />
+                {this.props.BingMapKey ? <DeviceMap {...deviceMapProps}/> : null}
```
Comment (maple10001) [inline src/components/pages/dashboard/dashboard.js]: could use { this.props.BingMapKey && <DeviceMap {...deviceMapProps}/> }

Comment (swetha571) [reply src/components/pages/dashboard/dashboard.js]: Both are same.

Comment (spryor) [inline src/components/pages/dashboard/dashboard.js]: Doesn't it make more sense to place the checks for the key in the device map component itself? Is that difficult for some reason?

Comment (swetha571) [reply src/components/pages/dashboard/dashboard.js]: Complete deviceMap is based on the BingMapkey whether it is static or actual key. Previously it used to be there but i removed last twodays back.

File: src/components/deviceMap/deviceMap.js [hunk 2]
```diff
@@ -16,14 +16,18 @@ import './deviceMap.css';
 class DeviceMap extends Component {
   constructor(props) {
     super(props);
-    window.loadMap = () => MapPane.init(this.props.BingMapKey);
-    this.state = {
-      mapCallbackComplete: false
+    window.loadMap = () => {
+      MapPane.init(this.props.BingMapKey);
+      if (this.props.BingMapKey && this.props.BingMapKey !== config.STATUS_CODES.STATIC) { this.showMap(this.props); }
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: If you are going to use `{}` place the contents on a new line. If you want to go one line, don't include the `{}`.

File: src/components/deviceMap/deviceMap.js [hunk 3]
```diff
@@ -16,14 +16,18 @@ import './deviceMap.css';
...
     };
+    this.state = { mapCallbackComplete: false };
   }
 
-  componentDidMount() {
-    if (this.props.BingMapKey && this.props.BingMapKey !== config.STATUS_CODES.STATIC) {
+  componentDidMount() { this.addScript(this.props); }
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: With the `{}`, place the contents on a new line.

File: src/components/deviceMap/deviceMap.js [hunk 4]
```diff
@@ -32,23 +36,21 @@ class DeviceMap extends Component {
...
+      !nextProps.BingMapKey
     ) {
       //the data is not loaded yet, return !
       return;
     }
-    if (nextProps.BingMapKey && nextProps.BingMapKey !== config.STATUS_CODES.STATIC) {
-      this.showMap(nextProps);
-    }
+    this.addScript(nextProps);
+    if (nextProps.BingMapKey && nextProps.BingMapKey !== config.STATUS_CODES.STATIC) { this.showMap(nextProps); }
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: same comment about the `{}`

File: src/components/deviceMap/deviceMap.js [hunk 5]
```diff
@@ -32,23 +36,21 @@ class DeviceMap extends Component {
...
   }
 
   showMap(props) {
     this.setState({ mapCallbackComplete: true });
     const { devices, telemetryByDeviceGroup, alarmList, actions } = props;
 
-    if (!devices || !telemetryByDeviceGroup || !alarmList) {
-      return;
-    }
+    if (!devices || !telemetryByDeviceGroup || !alarmList) { return; }
```
Comment (spryor) [inline src/components/deviceMap/deviceMap.js]: same comment about `{}`

**Pull request merged**